ui/src/actions/WebtilesPreviewer.js

```diff
@@ -29,97 +36,68 @@ export function selectType(tileType) {
-      isFetching: false
-    };
-  });
-  return {
-    type: RECEIVE_LOCALES,
-    channel: channel,
-    locales: locales,
-    receivedAt: Date.now()
-  };
+function checkStatus(response){
+  if (response.status >= 200 && response.status < 300) {
+    return response;
+  } else {
+    if (response.status === 404) {
+      throw new Error('No distribution found for this date.');
+    }
+    throw new Error('HTTP ' + response.status + ' - ' + response.statusText);
+  }
 }
 
-function receiveTiles(channel, locale, json) {
-  return {
-    type: RECEIVE_TILES,
-    channel: channel,
-    locale: locale,
-    directoryTiles: json.directory,
-    suggestedTiles: json.suggested,
-    receivedAt: Date.now()
-  };
-}
+export function fetchDistribution(moment) {
+  return (dispatch, getState) => {
+    dispatch((() => ({type: LOAD_DISTRIBUTION_START}))());
 
-function fetchLocales(state, channel) {
-  return dispatch => {
-    dispatch(requestLocales(channel));
-    return fetch(state.channels[channel].localeIndexUrl)
+    let url = getState().WebtilesPreviewer.distribution.fetchUrl;
+    url += '?date=' + moment.format('YYYY-MM-D')
+    return fetch(url)
+      .then(checkStatus)
       .then(response => response.json())
-      .then(json => dispatch(receiveLocales(channel, json)));
-  };
-}
+      .then(json => {
+        // Process the results
+        let channels = {};
+        Object.keys(json.results).forEach(function(channel) {
+          channels[channel] = {};
 
-function fetchTiles(state, channel, locale) {
-  return dispatch => {
-    dispatch(requestTiles(channel, locale));
-    return fetch(state.channels[channel].locales[locale].tileIndexUrl)
-      .then(response => response.json())
-      .then(json => dispatch(receiveTiles(channel, locale, json)));
-  };
-}
+          // Process the S3 artifacts into something more usable
+          let index;
+          let otherArtifacts = {};
+          json.results[channel].forEach(function(artifact) {
+            if (artifact.key.includes('index')) {
+              // The index file key will be something like: desktop_tile_index.v3.json
+              index = JSON.parse(artifact.data);
+            } else {
+              otherArtifacts[artifact.key] = JSON.parse(artifact.data);
+            }
+          });
 
-function shouldFetchLocales(state, channel) {
-  const channelObj = state.channels[channel];
-  if (!channelObj.locales && !channelObj.isFetching) {
-    return true;
-  }
-  return false;
-}
+          // Process all the locales in the index
+          Object.keys(index).forEach(function(locale) {
+            if (locale === '__ver__') return;  // skip that garbage
 
-function shouldFetchTiles(state, channel, locale) {
-  const localeObj = state.channels[channel].locales[locale];
-  if (localeObj && !localeObj.directoryTiles && !localeObj.isFetching) {
-    return true;
-  }
-  return false;
-}
+            // Get the artifact key
+            let agUrl = index[locale].ag;
```

Comment:
oh... Well let me just add a check here and for now will only use the first item in array. then will fix for real it in a new PR.